app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryListFragment.java

```diff
@@ -0,0 +1,306 @@
+        final List<InputMethodInfo> imis = imm.getEnabledInputMethodList();
+
+        for (final InputMethodInfo imi : imis) {
+            final List<InputMethodSubtype> subtypes = imm.getEnabledInputMethodSubtypeList(imi, true);
+            for (InputMethodSubtype subtype : subtypes) {
+                final String locale = subtype.getLocale();
+                if (!TextUtils.isEmpty(locale)) {
+                    localeSet.add(locale);
+                }
+            }
+        }
+
+        // We come here after we have collected locales from existing user dictionary entries and
+        // enabled subtypes. If we already have the locale-without-country version of the system
+        // locale, we don't add the system locale to avoid confusion even though it's technically
+        // correct to add it.
+        if (!localeSet.contains(Locale.getDefault().getLanguage())) {
+            localeSet.add(Locale.getDefault().toString());
+        }
+
+        return localeSet;
+    }
+
+    /**
+     * Creates the entries that allow the user to go into the user dictionary for each locale.
+     * @param userDictGroup The group to put the settings in.
+     */
+    protected void createUserDictSettings(final PreferenceGroup userDictGroup) {
+        final SharedPreferences prefs = DeviceProtectedUtils.getSharedPreferences(requireActivity().getApplicationContext());
+        final boolean localeSystemOnly = prefs.getBoolean(Settings.PREF_USE_SYSTEM_LOCALES, true);
+        // List of enabled user dictionary languages
+        final TreeSet<String> enabledUserDictionary = getUserDictionaryLocalesSet(requireActivity());
+        // List of main language
+        final List<InputMethodSubtype> enabledMainSubtype = SubtypeSettingsKt.getEnabledSubtypes(prefs, true);
+        // List of enabled system language
+        final List<Locale> enabledSystemLocale = SubtypeSettingsKt.getSystemLocales();
+        // To combine lists and handle duplicates automatically
+        LinkedHashSet<String> userDictionaryLanguage = new LinkedHashSet<>();
+
+        // Add "For all languages"
+        userDictionaryLanguage.add("");
+
+        // Add the language from the user dictionary if a word is present
+        if (enabledUserDictionary != null) {
+            userDictionaryLanguage.addAll(enabledUserDictionary);
+        }
+
+        // Add the main language selected in the "Language and Layouts" setting
+        for (InputMethodSubtype mainSubtype : enabledMainSubtype) {
+            Locale mainLocale = LocaleUtils.constructLocaleFromString(mainSubtype.getLocale());
+            String mainLocaleString = mainLocale.toString();
+            // Special treatment for the known languages with _ZZ and _zz types
+            if (mainLocaleString.endsWith("_ZZ") || mainLocaleString.endsWith("_zz") || mainLocaleString.equals("zz")) {
+                mainLocaleString = LocaleUtils.getLocaleDisplayNameInSystemLocale(mainLocale, requireActivity()).toLowerCase();
```

Comment:
This will will show _serbian (latin)_ instead of _Serbian (Latin)_. I don't think this is what you want to achieve with `toLowerCase`?

Reply:
I didn't notice this because on my phone the first letter of the languages (in the "Languages & Layouts" settings) is capitalized only if my phone is in English.
I have no idea why.

Reply:
@Helium314 Is there a reason why the first letters of each language are capitalized in the "Languages & Layouts" settings? Is it because it's encoded in kotlin?
If so, is it right to use this kind of code:

```java
private String capitalize(final String line) {
   return Character.toUpperCase(line.charAt(0)) + line.substring(1);
}
```

Reply:
> Is there a reason why the first letters of each language are capitalized in the "Languages & Layouts" settings?

The displayed name is `LocaleUtils.getLocaleDisplayNameInSystemLocale(locale, context)`, without any further changes. This (for most locales) ends up with `locale.getDisplayName(context.getResources().getConfiguration().locale)`, and what exactly this returns is determined by the Android system (and of course depends on the system locale as configures in the `Resources`).

Manually messing with the capitalization may have unexpected results, depending on the locales.

Reply:
Rectified with commit f950744